[Access] Crash when subscribe to "event" topic via new WebSockets

## Context

When trying to subscribe for events using the next message:
```json
{
    "action": "subscribe",
    "topic": "events"
}
```
Access node crash with next stacktrace:
```sh
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x22238a4]

goroutine 2788655 [running]:
github.com/onflow/flow-go/engine/access/state_stream/backend.(*StateStreamBackend).SubscribeEventsFromLatest(0x412ac5?, {0x3380290?, 0xc12462e910?}, {0x0, 0xc11948a5a0, 0xc11948a5d0, 0xc11948a600, 0xc11948a630})
        <autogenerated>:1 +0x44
github.com/onflow/flow-go/engine/access/rest/websockets/data_providers.(*EventsDataProvider).createSubscription(0xc102a9b480, {0x3380290, 0xc12462e910}, {{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...})
        /app/engine/access/rest/websockets/data_providers/events_provider.go:131 +0xfe
github.com/onflow/flow-go/engine/access/rest/websockets/data_providers.NewEventsDataProvider({0x3380290, 0xc11a4216d0}, {{0x336ca10, 0xc00142c440}, 0xff, {0x335bce0, 0xc000e4c320}, {0xc145af3800, 0x71, 0x1f4}, ...}, ...)
        /app/engine/access/rest/websockets/data_providers/events_provider.go:72 +0x48c
github.com/onflow/flow-go/engine/access/rest/websockets/data_providers.(*DataProviderFactoryImpl).NewDataProvider(0xc233ac7080?, {0x3380290, 0xc11a4216d0}, {0xc103130690, 0x78dd59?}, {0xc28edff50a, 0x38?}, 0xc128a40d78?, 0x78e07a?)
        /app/engine/access/rest/websockets/data_providers/factory.go:109 +0x24e
github.com/onflow/flow-go/engine/access/rest/websockets.(*Controller).handleSubscribe(0xc233ac7080, {0x3380290, 0xc11a4216d0}, {{{0x0, 0x0}, {0xc28edff500, 0x9}}, {0xc28edff50a, 0x6}, 0x0})
        /app/engine/access/rest/websockets/controller.go:429 +0x24c
github.com/onflow/flow-go/engine/access/rest/websockets.(*Controller).handleMessage(0xc233ac7080, {0x3380290, 0xc11a4216d0}, {0xc223f4ab00, 0x34, 0x40})
        /app/engine/access/rest/websockets/controller.go:375 +0x1c5
github.com/onflow/flow-go/engine/access/rest/websockets.(*Controller).readMessages(0xc233ac7080, {0x3380290, 0xc11a4216d0})
        /app/engine/access/rest/websockets/controller.go:350 +0xd6
github.com/onflow/flow-go/engine/access/rest/websockets.(*Controller).HandleConnection.func3()
        /app/engine/access/rest/websockets/controller.go:182 +0x1f
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /go/pkg/mod/golang.org/x/sync@v0.10.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 2788706
        /go/pkg/mod/golang.org/x/sync@v0.10.0/errgroup/errgroup.go:75 +0x96
```

--- a/engine/access/rest/websockets/controller.go
+++ b/engine/access/rest/websockets/controller.go
@@ -315,20 +315,22 @@ func (c *Controller) readMessages(ctx context.Context) error {
 				return err
 			}
 
+			err = wrapErr("error reading message", err)
 			c.writeErrorResponse(
 				ctx,
 				err,
-				wrapErrorMessage(http.StatusBadRequest, "error reading message", "", ""),
+				wrapErrorMessage(http.StatusBadRequest, err.Error(), "", ""),
 			)
 			continue
 		}
 
 		err := c.handleMessage(ctx, message)
 		if err != nil {
+			err = wrapErr("error parsing message", err)
 			c.writeErrorResponse(
 				ctx,
 				err,
-				wrapErrorMessage(http.StatusBadRequest, "error parsing message", "", ""),
+				wrapErrorMessage(http.StatusBadRequest, err.Error(), "", ""),
 			)
 			continue
 		}
@@ -374,11 +376,11 @@ func (c *Controller) handleMessage(ctx context.Context, message json.RawMessage)
 func (c *Controller) handleSubscribe(ctx context.Context, msg models.SubscribeMessageRequest) {
 	subscriptionID, err := c.parseOrCreateSubscriptionID(msg.SubscriptionID)
 	if err != nil {
+		err = wrapErr("error parsing subscription id", err)
 		c.writeErrorResponse(
 			ctx,
 			err,
-			wrapErrorMessage(http.StatusBadRequest, "error parsing subscription id",
-				models.SubscribeAction, msg.SubscriptionID),
+			wrapErrorMessage(http.StatusBadRequest, err.Error(), models.SubscribeAction, msg.SubscriptionID),
 		)
 		return
 	}
@@ -386,11 +388,11 @@ func (c *Controller) handleSubscribe(ctx context.Context, msg models.SubscribeMe
 	// register new provider
 	provider, err := c.dataProviderFactory.NewDataProvider(ctx, subscriptionID.String(), msg.Topic, msg.Arguments, c.multiplexedStream)
 	if err != nil {
+		err = wrapErr("error creating data provider", err)
 		c.writeErrorResponse(
 			ctx,
 			err,
-			wrapErrorMessage(http.StatusBadRequest, "error creating data provider",
-				models.SubscribeAction, subscriptionID.String()),
+			wrapErrorMessage(http.StatusBadRequest, err.Error(), models.SubscribeAction, subscriptionID.String()),
 		)
 		return
 	}
@@ -409,10 +411,11 @@ func (c *Controller) handleSubscribe(ctx context.Context, msg models.SubscribeMe
 	go func() {
 		err = provider.Run()
 		if err != nil {
+			err = wrapErr("internal error", err)
 			c.writeErrorResponse(
 				ctx,
 				err,
-				wrapErrorMessage(http.StatusInternalServerError, "internal error",
+				wrapErrorMessage(http.StatusInternalServerError, err.Error(),
 					models.SubscribeAction, subscriptionID.String()),
 			)
 		}
@@ -425,11 +428,11 @@ func (c *Controller) handleSubscribe(ctx context.Context, msg models.SubscribeMe
 func (c *Controller) handleUnsubscribe(ctx context.Context, msg models.UnsubscribeMessageRequest) {
 	subscriptionID, err := ParseClientSubscriptionID(msg.SubscriptionID)
 	if err != nil {
+		err = wrapErr("error parsing subscription id", err)
 		c.writeErrorResponse(
 			ctx,
 			err,
-			wrapErrorMessage(http.StatusBadRequest, "error parsing subscription id",
-				models.UnsubscribeAction, msg.SubscriptionID),
+			wrapErrorMessage(http.StatusBadRequest, err.Error(), models.UnsubscribeAction, msg.SubscriptionID),
 		)
 		return
 	}
@@ -512,6 +515,10 @@ func wrapErrorMessage(code int, message string, action string, subscriptionID st
 		},
 		Action: action,
 	}
+}
+
+func wrapErr(msg string, err error) error {
+	return errors.Join(fmt.Errorf("%s", msg), err)
 }
 
 func (c *Controller) parseOrCreateSubscriptionID(id string) (SubscriptionID, error) {

--- a/engine/access/rest/websockets/controller.go
+++ b/engine/access/rest/websockets/controller.go
@@ -315,7 +315,7 @@ func (c *Controller) readMessages(ctx context.Context) error {
 				return err
 			}
 
-			err = wrapErr("error reading message", err)
+			err = fmt.Errorf("error reading message: %w", err)
 			c.writeErrorResponse(
 				ctx,
 				err,
@@ -326,7 +326,7 @@ func (c *Controller) readMessages(ctx context.Context) error {
 
 		err := c.handleMessage(ctx, message)
 		if err != nil {
-			err = wrapErr("error parsing message", err)
+			err = fmt.Errorf("error parsing message: %w", err)
 			c.writeErrorResponse(
 				ctx,
 				err,
@@ -376,7 +376,7 @@ func (c *Controller) handleMessage(ctx context.Context, message json.RawMessage)
 func (c *Controller) handleSubscribe(ctx context.Context, msg models.SubscribeMessageRequest) {
 	subscriptionID, err := c.parseOrCreateSubscriptionID(msg.SubscriptionID)
 	if err != nil {
-		err = wrapErr("error parsing subscription id", err)
+		err = fmt.Errorf("error parsing subscription id: %w", err)
 		c.writeErrorResponse(
 			ctx,
 			err,
@@ -388,7 +388,7 @@ func (c *Controller) handleSubscribe(ctx context.Context, msg models.SubscribeMe
 	// register new provider
 	provider, err := c.dataProviderFactory.NewDataProvider(ctx, subscriptionID.String(), msg.Topic, msg.Arguments, c.multiplexedStream)
 	if err != nil {
-		err = wrapErr("error creating data provider", err)
+		err = fmt.Errorf("error creating data provider: %w", err)
 		c.writeErrorResponse(
 			ctx,
 			err,
@@ -411,7 +411,7 @@ func (c *Controller) handleSubscribe(ctx context.Context, msg models.SubscribeMe
 	go func() {
 		err = provider.Run()
 		if err != nil {
-			err = wrapErr("internal error", err)
+			err = fmt.Errorf("internal error: %w", err)
 			c.writeErrorResponse(
 				ctx,
 				err,
@@ -428,7 +428,7 @@ func (c *Controller) handleSubscribe(ctx context.Context, msg models.SubscribeMe
 func (c *Controller) handleUnsubscribe(ctx context.Context, msg models.UnsubscribeMessageRequest) {
 	subscriptionID, err := ParseClientSubscriptionID(msg.SubscriptionID)
 	if err != nil {
-		err = wrapErr("error parsing subscription id", err)
+		err = fmt.Errorf("error parsing subscription id: %w", err)
 		c.writeErrorResponse(
 			ctx,
 			err,
@@ -515,10 +515,6 @@ func wrapErrorMessage(code int, message string, action string, subscriptionID st
 		},
 		Action: action,
 	}
-}
-
-func wrapErr(msg string, err error) error {
-	return errors.Join(fmt.Errorf("%s", msg), err)
 }
 
 func (c *Controller) parseOrCreateSubscriptionID(id string) (SubscriptionID, error) {

--- a/engine/access/rest/websockets/data_providers/account_statuses_provider.go
+++ b/engine/access/rest/websockets/data_providers/account_statuses_provider.go
@@ -55,6 +55,12 @@ func NewAccountStatusesDataProvider(
 		heartbeatInterval: heartbeatInterval,
 	}
 
+	if p.stateStreamApi == nil {
+		logger.Debug().Msg("state stream api is nil. cannot stream account statuses")
+		return nil, fmt.Errorf("this acess node is configured with no state stream api. " +
+			"cannot stream account statuses")
+	}
+
 	// Initialize arguments passed to the provider.
 	accountStatusesArgs, err := parseAccountStatusesArguments(arguments, chain, eventFilterConfig)
 	if err != nil {

--- a/engine/access/rest/websockets/data_providers/account_statuses_provider.go
+++ b/engine/access/rest/websockets/data_providers/account_statuses_provider.go
@@ -49,16 +49,14 @@ func NewAccountStatusesDataProvider(
 	eventFilterConfig state_stream.EventFilterConfig,
 	heartbeatInterval uint64,
 ) (*AccountStatusesDataProvider, error) {
+	if stateStreamApi == nil {
+		return nil, fmt.Errorf("this access node does not support streaming account statuses")
+	}
+
 	p := &AccountStatusesDataProvider{
 		logger:            logger.With().Str("component", "account-statuses-data-provider").Logger(),
 		stateStreamApi:    stateStreamApi,
 		heartbeatInterval: heartbeatInterval,
-	}
-
-	if p.stateStreamApi == nil {
-		logger.Debug().Msg("state stream api is nil. cannot stream account statuses")
-		return nil, fmt.Errorf("this acess node is configured with no state stream api. " +
-			"cannot stream account statuses")
 	}
 
 	// Initialize arguments passed to the provider.

--- a/engine/access/rest/websockets/data_providers/account_statuses_provider_test.go
+++ b/engine/access/rest/websockets/data_providers/account_statuses_provider_test.go
@@ -105,7 +105,7 @@ func (s *AccountStatusesProviderSuite) TestAccountStatusesDataProvider_StateStre
 	)
 	s.Require().Nil(provider)
 	s.Require().Error(err)
-	s.Require().Contains(err.Error(), "cannot stream account statuses")
+	s.Require().Contains(err.Error(), "does not support streaming account statuses")
 }
 
 func (s *AccountStatusesProviderSuite) subscribeAccountStatusesDataProviderTestCases(

--- a/engine/access/rest/websockets/data_providers/account_statuses_provider_test.go
+++ b/engine/access/rest/websockets/data_providers/account_statuses_provider_test.go
@@ -85,6 +85,29 @@ func (s *AccountStatusesProviderSuite) TestAccountStatusesDataProvider_HappyPath
 	)
 }
 
+func (s *AccountStatusesProviderSuite) TestAccountStatusesDataProvider_StateStreamNotConfigured() {
+	ctx := context.Background()
+	send := make(chan interface{})
+
+	topic := AccountStatusesTopic
+
+	provider, err := NewAccountStatusesDataProvider(
+		ctx,
+		s.log,
+		nil,
+		"dummy-id",
+		topic,
+		models.Arguments{},
+		send,
+		s.chain,
+		state_stream.DefaultEventFilterConfig,
+		subscription.DefaultHeartbeatInterval,
+	)
+	s.Require().Nil(provider)
+	s.Require().Error(err)
+	s.Require().Contains(err.Error(), "cannot stream account statuses")
+}
+
 func (s *AccountStatusesProviderSuite) subscribeAccountStatusesDataProviderTestCases(
 	backendResponses []*backend.AccountStatusesResponse,
 ) []testType {

--- a/engine/access/rest/websockets/data_providers/events_provider.go
+++ b/engine/access/rest/websockets/data_providers/events_provider.go
@@ -54,6 +54,11 @@ func NewEventsDataProvider(
 		heartbeatInterval: heartbeatInterval,
 	}
 
+	if p.stateStreamApi == nil {
+		logger.Debug().Msg("state stream api is nil. cannot stream events")
+		return nil, fmt.Errorf("this acess node is configured with no state stream api. cannot stream events")
+	}
+
 	// Initialize arguments passed to the provider.
 	eventArgs, err := parseEventsArguments(arguments, chain, eventFilterConfig)
 	if err != nil {

--- a/engine/access/rest/websockets/data_providers/events_provider.go
+++ b/engine/access/rest/websockets/data_providers/events_provider.go
@@ -48,15 +48,14 @@ func NewEventsDataProvider(
 	eventFilterConfig state_stream.EventFilterConfig,
 	heartbeatInterval uint64,
 ) (*EventsDataProvider, error) {
+	if stateStreamApi == nil {
+		return nil, fmt.Errorf("this access node does not support streaming events")
+	}
+
 	p := &EventsDataProvider{
 		logger:            logger.With().Str("component", "events-data-provider").Logger(),
 		stateStreamApi:    stateStreamApi,
 		heartbeatInterval: heartbeatInterval,
-	}
-
-	if p.stateStreamApi == nil {
-		logger.Debug().Msg("state stream api is nil. cannot stream events")
-		return nil, fmt.Errorf("this acess node is configured with no state stream api. cannot stream events")
 	}
 
 	// Initialize arguments passed to the provider.

--- a/engine/access/rest/websockets/data_providers/events_provider_test.go
+++ b/engine/access/rest/websockets/data_providers/events_provider_test.go
@@ -299,6 +299,29 @@ func (s *EventsProviderSuite) TestEventsDataProvider_InvalidArguments() {
 	}
 }
 
+func (s *EventsProviderSuite) TestEventsDataProvider_StateStreamNotConfigured() {
+	ctx := context.Background()
+	send := make(chan interface{})
+
+	topic := EventsTopic
+
+	provider, err := NewEventsDataProvider(
+		ctx,
+		s.log,
+		nil,
+		"dummy-id",
+		topic,
+		models.Arguments{},
+		send,
+		s.chain,
+		state_stream.DefaultEventFilterConfig,
+		subscription.DefaultHeartbeatInterval,
+	)
+	s.Require().Nil(provider)
+	s.Require().Error(err)
+	s.Require().Contains(err.Error(), "cannot stream events")
+}
+
 // invalidArgumentsTestCases returns a list of test cases with invalid argument combinations
 // for testing the behavior of events data providers. Each test case includes a name,
 // a set of input arguments, and the expected error message that should be returned.

--- a/engine/access/rest/websockets/data_providers/events_provider_test.go
+++ b/engine/access/rest/websockets/data_providers/events_provider_test.go
@@ -319,7 +319,7 @@ func (s *EventsProviderSuite) TestEventsDataProvider_StateStreamNotConfigured() 
 	)
 	s.Require().Nil(provider)
 	s.Require().Error(err)
-	s.Require().Contains(err.Error(), "cannot stream events")
+	s.Require().Contains(err.Error(), "does not support streaming events")
 }
 
 // invalidArgumentsTestCases returns a list of test cases with invalid argument combinations

